sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/notifications/fcm/PushNotificationClient.java

```diff
@@ -235,7 +262,42 @@ private ChatEvent parsePushNotificationEvent(ChatEventType chatEventType, ChatPu
         return NotificationUtils.toEventPayload(chatEventType, decrypted);
     }
 
-    private void scheduleRegistrationRenew(final long delayMs, final int tryCount) {
+    private void startRegistrationRenewalWorker(long intervalInMs) {
+        this.logger.info("Initialize RenewTokenWorker in background");
+        this.workManager = WorkManager.getInstance();
+        this.renewalWorkerDataContainer = RenewalWorkerDataContainer.instance();
+        Data inputData = new Data.Builder().putString("deviceRegistrationToken", deviceRegistrationToken).build();
+
+        PeriodicWorkRequest renewTokenRequest =
+            new PeriodicWorkRequest.Builder(RenewTokenWorker.class, intervalInMs, TimeUnit.MILLISECONDS)
+                .setBackoffCriteria(BackoffPolicy.LINEAR, 10, TimeUnit.SECONDS)
+                .setInputData(inputData)
+                .build();
+        //Ensure only one work executed.
+        workManager.cancelAllWork();
+        workManager.enqueue(renewTokenRequest);
+
+        //Checking the result of last execution. There are two states for periodic work: ENQUEUED and RUNNING. We do checking
+        //for every ENQUEUED state, meaning last execution has completed.
+        workManager.getWorkInfoByIdLiveData(renewTokenRequest.getId()).observeForever(workInfo -> {
+            if (workInfo != null && workInfo.getState() == WorkInfo.State.ENQUEUED) {
+                boolean failed = renewalWorkerDataContainer.isExecutionFail();
+                if (failed) {
+                    RuntimeException exception = new RuntimeException(
+                        "Registration renew request failed after "
+                            + NotificationUtils.MAX_REGISTRATION_RETRY_COUNT
+                            + "retries.");
+                    this.registrationErrorHandler.accept(exception);
+                    stopPushNotifications();
+                    this.logger.info("Renew token failed");
+                } else {
+                    this.logger.info("Renew token succeeded");
```

Comment:
If we use the class `ClientLogger` here it could mean that when an app is closed and re-opened we'd potentially have a second logger until the worker finally stops, right @anuchandy? I guess it wouldn't be a big deal as long as we don't create more than one worker for this type of work, but I'd still like to hear your opinion on this.